File: Support/Templates/vb/Files/EditableChildList.vb
Imports Csla.Security

<Serializable()> _
Public Class EditableChildList
  Inherits BusinessListBase(Of EditableChildList, EditableChild)

  <FetchChild>
  Private Sub Fetch(ByVal childData As Object)
    RaiseListChangedEvents = False
    For Each child As Object In CType(childData, IList(Of Object))
      Me.Add(DataPortal.FetchChild(Of EditableChild)(child))
    Next
    RaiseListChangedEvents = True
  End Sub

End Class

File: Support/Templates/cs/Files/EditableChildList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Templates
{
  [Serializable]
  public class EditableChildList : 
    BusinessListBase<EditableChildList, EditableChild>
  {
    [FetchChild]
    private void Fetch(object childData)
    {
      RaiseListChangedEvents = false;
      foreach (var child in (IList<object>)childData)
        this.Add(DataPortal.FetchChild<EditableChild>(child));
      RaiseListChangedEvents = true;
    }
  }
}
